### ISSUE TITLE
[3.1] wrong canonical url with a sub-folder url

### ISSUE THREAD

**Issue by hiousi** [bolt/bolt#5924]
hello,

With a brand new install of bolt with  `example.com/folder` rather than `subdom.example.com`

The header of HTML gives a link to canonical url that is ok for home page and  search page.

It is not the good one for a content page:
`example.com/folder/showcase/blub` has canonical url  `example.com/folder/folder/showcase/blub`

**Comment by bobdenotter** [comment]
Which version of Bolt is this on?

**Comment by hiousi** [comment]
bolt 3.1.5
this is what I found walking around the code:

ResourceManager::initializeRequest() set the good canonicalurl param.

Frontend::record() is doing wrong on line 167. 
` $url = $this->app['resources']->getUrl('rooturl') . ltrim($content->link(), '/');`

root url is `example.com/folder` and 
$content->link() is  `/folder/showcase/blub`

I am new to this great CMS and I may have missed something with my configuration of Bolt

**Comment by hiousi** [comment]
just upgraded to 3.2rc1 problem is still there.

**Comment by hiousi** [comment]
I think a wrong canonical url is bad for search engines. is there a simple workaround? can I disable canonical URL in the header for now?

**Comment by hiousi** [comment]
@bobdenotter can you help?

**Comment by stormfly-pl** [comment]
In my opinion problem is here :

```
    // Bolt\Provider\AssetServiceProvider
    return new PathPackage(
        $app['resources']->getUrl($name),
        $app['asset.version_strategy']($name),
        $app['asset.context'] // <- here
    );
```

**here** we have empty Symfony\Component\HttpFoundation\RequestStack after setup, getUrl() works good. BUT! Next application push there new object Symfony\Component\HttpFoundation\Request and that destroy everything. That's why unit tests are green ;)

**Comment by hiousi** [comment]
It seems that I am alone with this issue :(

**Comment by SahAssar** [comment]
Fixed in #6212, closing, and I'm very sorry for us being so tardy in giving this issue any attention and you having to open #6200 too to get this fixed :(
